src/jshint.js

```diff
@@ -3754,18 +3754,25 @@ var JSHINT = (function() {
     var b;
 
     function doCatch() {
-      var e;
+      var params = [];
 
       advance("catch");
       advance("(");
 
-      e = state.tokens.next;
-      if (e.type !== "(identifier)") {
-        warning("E030", state.tokens.next, e.value);
-        e = null;
+      if (checkPunctuators(state.tokens.next, ["[", "{"])) {
+        var tokens = destructuringExpression();
+        var t;
+        for (t in tokens) {
+          t = tokens[t];
+          if (t.id) {
+            params.push({ id: t.id, token: t });
+          }
+        }
+      } else if (state.tokens.next.type !== "(identifier)") {
+        warning("E030", state.tokens.next, state.tokens.next.value);
       } else {
         // only advance if we have an identifier so we can continue parsing in the most common error - that no param is given.
-        advance();
+        params.push({ id: identifier(), token: state.tokens.curr, type: "exception" });
```

Comment:
using identifier does allow the rest operator, which doesn't look like its allowed in exception catch.
This is already a bug elsewhere, but I am not sure at the moment whether identifier allowing spread is causing the allowance of spread everywhere or if its down to allowing "..." as an infix.
It might be nice to be able to pass something to identifier to tell it not to allow spread? not sure.